pkg/cache/sql/store/store.go

```diff
@@ -107,7 +120,7 @@ func (s *Store) upsert(key string, obj any) error {
 
 	err = s.Upsert(tx, s.upsertStmt, key, obj, s.shouldEncrypt)
 	if err != nil {
-		return err
+		return fmt.Errorf("while executing query: %s got error: %w", s.upsertQuery, err)
```

Comment:
This looks like it could be an error type, similar to https://pkg.go.dev/io/fs#PathError ?

```go
type QueryError struct {
  QueryString string
  Err error
}

func (e *QueryError) Error() string { return "while executing query: " + e.QueryString + " got error: " + e.Err.Error() }

func (e *QueryError) Unwrap() error { return e.Err }
```

Reply:
Right, thanks for the observation. Unfortunately commits in this PR were merged together as part of #84, so I am starting [a separate PR](https://github.com/rancher/lasso/pull/90) to address this feedback